src/Mod/Sketcher/CMakeLists.txt

```diff
@@ -35,7 +35,10 @@ set(Sketcher_Profiles
 add_custom_target(SketcherScripts ALL
     SOURCES ${Sketcher_Scripts} ${Sketcher_Profiles} ${Sketcher_TestScripts}
 )
-
+include_directories(
+        SYSTEM
+        ${EIGEN3_INCLUDE_DIR}
+)
```

Comment:
Since you're not changing Sketcher at all, I wouldn't expect this change to be required.